### ISSUE TITLE
No way to run commands/customizations after import, but before booting

### ISSUE THREAD

**Issue by mrkafk** [hashicorp/vagrant#2939]
It's often necessary to run some command or customization after the base box has been imported, but before first boot, e.g. to clean up stale NAT portforwarding, or attach new disks, etc.

However, it seems that for now Vagrantfile's "Vagrant.configure do |config| ... end" section runs BEFORE importing base box, hence ":id" symbol is empty and customizations cannot be done and vboxmanage cmds cannot be ran at this stage.

**Comment by realyst2k** [comment]
This is happening for me too on OSX Mavericks running VirtualBox 4.3.28.

```
  config.vm.network "private_network", ip: "192.168.123.4", auto_config: false
  config.vm.provider :virtualbox do |vb|
      vb.customize ["modifyvm", :id, "--natdnshostresolver1", "on"]
```

produces the same errror:

```
There was an error while executing `VBoxManage`, a CLI used by Vagrant
for controlling VirtualBox. The command and stderr is shown below.

Command: ["modifyvm", "8f287126-ed7a-4eec-b3fa-01098e7939b2", "--natpf1", "ssh,tcp,127.0.0.1,2222,,22"]

Stderr: VBoxManage: error: A NAT rule of this name already exists
```

I ran it with ---debug, lots of stuff in there, but I can send you all of it in txt form(with some strategic things ###'d out for business purposes) if you tell me where to send/upload it.  This appears to be where the error first pops up:

```
vcpfile="/Users/#######/VirtualBox VMs/######################_default_1432745920601_32311/######################_default_1432745920601_32311.webm"
vcpwidth=1024
vcpheight=768
vcprate=512
vcpfps=25
GuestMemoryBalloon=0
DEBUG subprocess: Waiting for process to exit. Remaining to timeout: 32000
DEBUG subprocess: Exit status: 0
 INFO interface: detail: 22 => 2222 (adapter 1)
 INFO interface: detail:     default: 22 => 2222 (adapter 1)
    default: 22 => 2222 (adapter 1)
 INFO subprocess: Starting process: ["/usr/bin/VBoxManage", "modifyvm", "fe3b50d9-5817-4d0b-9819-9e5545baba23", "--natpf1", "ssh,tcp,127.0.0.1,2222,,22"]
DEBUG subprocess: Command not in installer, not touching env vars.
DEBUG subprocess: Selecting on IO
DEBUG subprocess: stderr: VBoxManage: error: A NAT rule of this name already exists
VBoxManage: error: Details: code NS_ERROR_INVALID_ARG (0x80070057), component NATEngine, interface INATEngine, callee nsISupports
VBoxManage: error: Context: "AddRedirect(Bstr(strName).raw(), proto, Bstr(strHostIp).raw(), RTStrToUInt16(strHostPort), Bstr(strGuestIp).raw(), RTStrToUInt16(strGuestPort))" at line 1708 of file VBoxManageModifyVM.cpp
DEBUG subprocess: Waiting for process to exit. Remaining to timeout: 32000
DEBUG subprocess: Exit status: 1
```

And then it proceeds to do that again numerous times it looks like before giving up.  That block above is also the first instance of --natpf1 I could find in the entire log so I'm unsure what it's running into.

Basically, if I vagrant destroy, then vagrant up again, the first vagrant up produces this error.  The next vagrant up command works.

This is consistent each and every time.  Even when I nuke the .vagrant.d/boxes/#BOXNAME# directory.
